### ISSUE TITLE
Wiki-style linking

### ISSUE THREAD

**Issue by tomer** [jekyll/jekyll#4984]
While Markdown link command is great, I am wondering if it possible to have yet another way of linking to content: wiki-style links. This way, each page could have a unique ID, and each other page could link to that page by simply mentioning its `[[id]]` somewhere in the content. 

Wiki-style links are different that regular links: While it is possible to provide relative links to the target page, links will break if the source or target pages change location, and Jekyll has no way to warn the user of link breakage. On the other hand, IDs can be short and can be expanded to relative links on build time, so users won't have to worry about breaking internal links.

This can be simulated by [Data Files], but will require some custom programming, while enclosing page ID in doubled square brackets is way quicker. 

[Data Files]: https://jekyllrb.com/docs/datafiles/

**Comment by parkr** [comment]
Very cool! This is not something we will implement at the Jekyll level. Instead, this would be integrated into `kramdown` or similar. I believe this has been suggested/requested before, and we suggested this same remedy too.

Additionally, links like `[[id]]` are difficult to read for newcomers to a given page because it's not obvious what the page contains. For links like `[[title]]`, we end up with a lot of annoying usability issues because titles often change.

In short, a good ol' `[markdown](/link/)` is our best bet here. 😉 

Thanks!
